pkg/plugin/javascript/dashclient_test.go

```diff
@@ -0,0 +1,100 @@
+package javascript
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/vmware-tanzu/octant/pkg/store"
+	storeFake "github.com/vmware-tanzu/octant/pkg/store/fake"
+)
+
+func TestDashboardClient_Get(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	ctx := context.TODO()
+	objectStore := storeFake.NewMockStore(controller)
+
+	vm := goja.New()
+	vm.SetFieldNameMapper(goja.TagFieldNameMapper("json", true))
+
+	obj := CreateDashClientObject(ctx, objectStore, vm)
+	assert.NotNil(t, obj)
+
+	objectStore.EXPECT().Get(ctx, store.Key{APIVersion: "v1", Kind: "Pod"}).Return(&unstructured.Unstructured{}, nil)
+
+	vm.Set("dashClient", obj)
+	vm.RunString(`
+dashClient.Get({apiVersion: 'v1', kind:'Pod'})
+`)
+}
+
+func TestDashboardClient_List(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	ctx := context.TODO()
+	objectStore := storeFake.NewMockStore(controller)
+
+	vm := goja.New()
+	vm.SetFieldNameMapper(goja.TagFieldNameMapper("json", true))
+
+	obj := CreateDashClientObject(ctx, objectStore, vm)
+	assert.NotNil(t, obj)
+
+	objectStore.EXPECT().List(ctx, store.Key{Namespace: "test", APIVersion: "v1", Kind: "Pod"}).Return(&unstructured.UnstructuredList{}, false, nil)
+
+	vm.Set("dashClient", obj)
+	vm.RunString(`
+dashClient.List({namespace:'test', apiVersion: 'v1', kind:'Pod'})
+`)
+}
+
+func TestDashboardClient_CreateUpdate(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	ctx := context.TODO()
```

Comment:
This is a nit: context.TODO() is for when you are going to plumb a context in later. Use context.Background() instead. They are literally the same thing, but only differ intent.